src/Form/DataTransformer/PhoneNumberToArrayTransformer.php

```diff
@@ -100,4 +114,35 @@ public function reverseTransform(mixed $value): ?PhoneNumber
 
         return $phoneNumber;
     }
+
+    private function removeLeadingZeros(string $number, int $leadingZeros): string
+    {
+        if ($leadingZeros > 0) {
+            $pattern = '/^0{'.$leadingZeros.'}/';
+
+            /* @phpstan-ignore-next-line */
+            return preg_replace($pattern, '', $number);
+        }
+
+        return $number;
+    }
+
+    private function addLeadingZeros(string $number, string $country): string
+    {
+        $metadata = $this->phoneNumberUtil->getMetadataForRegion($country);
+
+        if ($metadata && $metadata->getGeneralDesc()->hasNationalNumberPattern()) {
+            $pattern = $metadata->getGeneralDesc()->getNationalNumberPattern();
+
+            // Determine the number of leading zeros to add
+            if (preg_match('/^(0+)/', $pattern, $matches)) {
+                $leadingZeros = $matches[1];
+                if (!str_starts_with($number, $leadingZeros)) {
+                    $number = $leadingZeros.$number;
+                }
+            }
+        }
+
+        return $number;
```

Comment:
Could you refactor this function using [return early pattern](https://medium.com/swlh/return-early-pattern-3d18a41bba8) please (see suggestion above)? :pray:

Reply:
I don't know why I didn't do that in the first place 😅